internal/core/application/service.go

```diff
@@ -2177,43 +2192,105 @@ func (s *Service) refundVHTLC(
 		return s.SignTransaction(ctx, encoded)
 	}
 
+	// user signing
 	signedRefundTx, err := signTransaction(refundTx)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to sign refund tx: %s", err)
+	}
+	signedCheckpointTx, err := signTransaction(checkpointPtxs[0])
+	if err != nil {
+		return "", fmt.Errorf("failed to sign refund tx: %s", err)
 	}
 
+	signedRefundPsbt, err := psbt.NewFromRawBytes(strings.NewReader(signedRefundTx), true)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode refund tx signed by us: %s", err)
+	}
+
+	signedCheckpointPsbt, err := psbt.NewFromRawBytes(strings.NewReader(signedCheckpointTx), true)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode refund tx signed by us: %s", err)
+	}
+
+	finalRefundUpdater, err := psbt.NewUpdater(signedRefundPsbt)
+	if err != nil {
+		return "", fmt.Errorf("failed to init updater for final refund tx: %s", err)
+	}
+
+	checkpointsList := make([]*psbt.Packet, 0)
+
+	checkpointsList = append(checkpointsList, signedCheckpointPsbt)
+
 	if withReceiver {
-		signedRefundTx, err = s.boltzRefundSwap(swapId, signedRefundTx)
+		boltzSignedRefundTx, boltzSignedCheckpointTx, err := s.boltzRefundSwap(
+			swapId, unsignedRefundTx, unsignedCheckpointTx,
+		)
 		if err != nil {
 			return "", err
 		}
-	}
 
-	checkpointTxs := make([]string, 0, len(checkpointPtxs))
-	for _, ptx := range checkpointPtxs {
-		tx, err := ptx.B64Encode()
+		boltzSignedRefundPsbt, err := psbt.NewFromRawBytes(strings.NewReader(boltzSignedRefundTx), true)
 		if err != nil {
-			return "", err
+			return "", fmt.Errorf("failed to decode refund tx signed by boltz: %s", err)
 		}
-		checkpointTxs = append(checkpointTxs, tx)
+		boltzSignedCheckpointPsbt, err := psbt.NewFromRawBytes(
+			strings.NewReader(boltzSignedCheckpointTx), true,
+		)
+		if err != nil {
+			return "", fmt.Errorf("failed to decode refund tx signed by boltz: %s", err)
+		}
+
+		for i := range finalRefundUpdater.Upsbt.Inputs {
+			boltzIn := boltzSignedRefundPsbt.Inputs[i]
+			partialSig := boltzIn.PartialSigs[0]
```

Comment:
```suggestion
			partialSig := boltzIn.TaprootScriptSpendSig[0]
```

we should combine TaprootScriptSpendSig (field for tapscript signatures), not PartialSigs (field for non-taproot sigs)